packages/core/strapi/lib/middlewares/logger/index.js

```diff
@@ -24,38 +23,21 @@ module.exports = strapi => {
      * Initialize the hook
      */
     initialize() {
-      const { level, exposeInContext, requests } = strapi.config.middleware.settings.logger;
+      const { exposeInContext, requests } = strapi.config.middleware.settings.logger;
 
-      const logLevels = Object.keys(strapi.log.levels.values);
-
-      if (!_.includes(logLevels, level)) {
-        throw new Error(
-          "Invalid log level set in middleware configuration. Accepted values are: '" +
-            logLevels.join("', '") +
-            "'."
-        );
-      }
-
-      strapi.log.level = level;
+      const { levels: logLevels, level: currentLogLevel } = strapi.log;
 
       if (exposeInContext) {
         strapi.app.context.log = strapi.log;
       }
 
-      const isLogLevelEnvVariableSet = _.isString(process.env.STRAPI_LOG_LEVEL);
-
-      if (isLogLevelEnvVariableSet && strapi.log.levelVal <= 20) {
-        strapi.log.debug(
-          `STRAPI_LOG_LEVEL environment variable is overridden by logger middleware. It only applies outside Strapi's middleware context.`
-        );
-      }
-
-      if (requests && strapi.log.levelVal <= 20) {
+      if (requests && currentLogLevel >= logLevels.debug) {
```

Comment:
I think we could remove this to simplify. if the user changes the logger log level it will remove those logs accordingly. Otherwise the dev can just disable this middleware. wdyt ?

Reply:
lgtm